Classes/common/CDTFetchChanges.m

```diff
@@ -91,7 +100,10 @@ - (SequenceNumber)notifyChanges:(TD_RevisionList *)changes
     }
```

Comment:
This loops only gets a maximum of 500 documents as the call to get the list of database changes uses blocks of 500, which are then passed to this method.

As a second point, this code isn't used for replications.